pkg/operator/nsfinalizercontroller/finalizer_controller.go

```diff
@@ -79,14 +84,22 @@ func (c finalizerController) sync() error {
 		return nil
 	}
 
-	pods, err := c.podLister.Pods(operatorclient.TargetNamespace).List(labels.Everything())
+	// allow one minute of grace for most things to terminate.
+	// TODO now that we have conditions, we may be able to check specific conditions
+	deletedMoreThanAMinute := ns.DeletionTimestamp.Time.Add(1 * time.Minute).Before(time.Now())
+	if !deletedMoreThanAMinute {
+		c.queue.AddAfter(c.namespaceName, 1*time.Minute)
+		return nil
+	}
+
+	pods, err := c.podLister.Pods(c.namespaceName).List(labels.Everything())
```

Comment:
@deads2k can we narrow this list to something that only select apiserver pods?

Reply:
> @deads2k can we narrow this list to something that only select apiserver pods?

the point of this block is to ensure that all the pods are gone.  Scoping it down defeats the purpose.